azure-functions-codegen/src/func.rs

```diff
@@ -83,6 +85,44 @@ fn validate_orchestration_function(func: &ItemFn) {
     }
 }
 
+fn validate_activity_function(func: &ItemFn) {
+    // Activity functions cannot have a $return binding
+    // Default, -> ActivityOutput, and -> (ActivityOutput, ...) are acceptable
+
+    fn validate_return_binding(ty: &Type) {
+        match ty {
+            Type::Tuple(tuple) => {
+                if let Some(first) = tuple.elems.iter().nth(0) {
+                    validate_return_binding(first)
+                }
+            }
+            Type::Paren(tp) => validate_return_binding(&*tp.elem),
+            Type::Path(tp) => {
+                if last_segment_in_path(&tp.path).ident != ACTIVITY_OUTPUT_TYPE {
+                    macro_panic(
+                        tp.span(),
+                        format!(
+                            "activity functions must have a return type of `{}`",
+                            ACTIVITY_OUTPUT_TYPE
+                        ),
+                    );
+                }
+            }
+            _ => macro_panic(
+                ty.span(),
+                format!(
+                    "activity functions must have a return type of `{}`",
+                    ACTIVITY_OUTPUT_TYPE
+                ),
+            ),
+        }
+    }
+
+    if let ReturnType::Type(_, ty) = &func.decl.output {
```

Comment:
RE: the moment of confusion of what `_` represents, could it be useful to rename this to `_span` so we don't forget in 4 months when Durable Functions 2: Electric Boogaloo comes out?

Reply:
I prefer `_` simply because the code doesn't really care what it represents.  Pulling up the `syn::ReturnType::Type` documentation isn't difficult if someone is really curious.